clients/rge/index.ts

```diff
@@ -0,0 +1,127 @@
+import routes from '#clients/routes';
+import {
+  IRGECompanyCertifications,
+  NomCertificat,
+} from '#models/garant-environement';
+import { Siren, Siret } from '#utils/helpers';
+import { httpGet } from '#utils/network';
+
+/*
+ * Reconnu Garant de l'Environnement (RGE)
+ * https://france-renov.gouv.fr/annuaire-rge
+ */
+
+export enum WORKING_WITH_ENUM {
+  PROFESSIONAL = 'PROFESSIONAL',
+  INDIVIDUAL = 'INDIVIDUAL',
+}
+
+interface IRGEResponse {
+  results: {
+    adresse: string;
+    code_postal: string;
+    code_qualification: string;
+    commune: string;
+    domaine: string;
+    email: string;
+    nom_certificat: NomCertificat;
+    nom_entreprise: string;
+    nom_qualification: string;
+    organisme: string;
+    particulier: Boolean;
+    siret: Siret;
+    site_internet: string;
+    telephone: string;
+    url_qualification: string;
+  }[];
+}
+
+const clientRGE = async (
+  siren: Siren,
+  useCache = true
+): Promise<IRGECompanyCertifications | {}> => {
+  const route = routes.rge.lines;
+  const response = await httpGet(
+    route,
+    { params: { qs: `siret:${siren}*` } },
+    useCache
+  );
+
+  return mapToDomainObject(response.data as IRGEResponse);
+};
+
+export const mapToDomainObject = (
+  rge: IRGEResponse
+): IRGECompanyCertifications => {
+  if (!rge.results.length) {
+    return {
+      companyInfo: null,
+      certifications: [],
+    };
+  }
+
+  const [firstResult] = rge.results;
+
+  const {
+    adresse,
+    code_postal,
+    commune,
+    nom_entreprise,
+    particulier,
+    siret,
+    telephone,
+    site_internet,
+    email,
+  } = firstResult;
+
+  const companyInfo: IRGECompanyCertifications['companyInfo'] = {
+    adresse: `${adresse}, ${code_postal}, ${commune}`,
+    email,
+    nomEntreprise: nom_entreprise,
+    siret: siret,
+    siteInternet: site_internet,
+    telephone: telephone,
+    workingWith: [
+      WORKING_WITH_ENUM.PROFESSIONAL,
+      ...(particulier ? [WORKING_WITH_ENUM.INDIVIDUAL] : []),
+    ],
+  };
+
+  const certifications: IRGECompanyCertifications['certifications'] = [];
+
+  rge.results.forEach((result) => {
```

Comment:
I find this part is a quite complicated. I would rather do a reduce returning an object with `nom_certificat` as key and 
```
{
        code_qualification,
        domaine,
        nom_certificat,
        nom_qualification,
        organisme,
        url_qualification,
      }
```
as values. Then use `Object.values()` ot get your `certifications`

Reply:
Why not we can talk about it in visio